torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -439,7 +439,28 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
           if (futureResponse->completed()) {
             --serverActiveCalls_;
             if (!futureResponse->hasError()) {
-              send(work.from_, std::move(*futureResponse).moveValue());
+              // Hack: Check for device type here since throwing an exception in
+              // the send() below is not handled by enqueueRecv yet. Should be
+              // fixed as part of
+              // https://github.com/pytorch/pytorch/issues/25516.
+              const auto& respMessage = futureResponse->wait();
+              std::string errorMsg;
+              for (const auto& tensor : respMessage.tensors()) {
+                if (!tensor.device().is_cpu()) {
+                  errorMsg = c10::str(
+                      "ProcessGroup RPC backend only supports",
+                      " CPU tensors, please move your tensors to CPU before sending ",
+                      "them over RPC. Found tensor on device: ",
```

Comment:
Shall we make this error message different from the one in `wireSerialize`? If we got here, it seems we hit a bigger problem.

Reply:
Its a similar issue as wireSerialize, the only difference here is that the Tensor is being sent as a response to an RPC.